src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -165,155 +292,847 @@ private MessageConsumer newReplyStream(
+                    hasMore = true;
+                }
+                AmqpFrameHeaderFW updatedFrameHeader = amqpFrameHeaderRW.wrap(frameBuffer, 0, frameBuffer.capacity())
+                    .size(frameHeaderSize + transfer.sizeof() + messageSize + payloadSize)
+                    .doff(2)
+                    .type(0)
+                    .channel(channel)
+                    .performative(b -> b.domainId(d -> d.set(DESCRIBED))
+                        .descriptorId(d2 -> d2.set(AmqpFrameType.TRANSFER.value())))
+                    .build();
+                payloadRW.put(updatedFrameHeader.buffer(), updatedFrameHeader.offset(), updatedFrameHeader.sizeof())
+                    .put(transfer.buffer(), transfer.offset(), transfer.sizeof())
+                    .put(messageInfo.buffer(), messageInfo.offset(), messageInfo.sizeof());
+            }
+            else
+            {
+                if (payloadIndex + (initialMaxFrameSize - frameHeaderSize - transferFrameSize) > payload.sizeof())
+                {
+                    payloadSize = payload.sizeof() - payloadIndex;
+                }
+                else
+                {
+                    transfer = amqpTransferRW.wrap(frameBuffer, transferOffset, frameBuffer.capacity())
+                        .handle(handle)
+                        .deliveryId(dataEx.deliveryId())
+                        .deliveryTag(deliveryTag)
+                        .messageFormat(dataEx.messageFormat())
+                        .settled(settled)
+                        .more(1)
+                        .build();
+                    payloadSize = initialMaxFrameSize - frameHeaderSize - transferFrameSize;
+                    hasMore = true;
+
+                }
+                AmqpFrameHeaderFW updatedFrameHeader = amqpFrameHeaderRW.wrap(frameBuffer, 0, frameBuffer.capacity())
+                    .size(frameHeaderSize + transfer.sizeof() + payloadSize)
+                    .doff(2)
+                    .type(0)
+                    .channel(channel)
+                    .performative(b -> b.domainId(d -> d.set(DESCRIBED))
+                        .descriptorId(d2 -> d2.set(AmqpFrameType.TRANSFER.value())))
+                    .build();
+                payloadRW.put(updatedFrameHeader.buffer(), updatedFrameHeader.offset(), updatedFrameHeader.sizeof())
+                    .put(transfer.buffer(), transfer.offset(), transfer.sizeof());
+            }
+
+            payloadRW.put(payload.buffer(), payload.offset() + payloadIndex, payloadSize);
+            OctetsFW transferWithFrameHeader = payloadRW.build();
+            doNetworkData(traceId, authorization, 0L, transferWithFrameHeader);
+
+            if (hasMore)
+            {
+                doEncodeTransfer(traceId, authorization, channel, handle, flags, extension, payload,
+                    payloadIndex + payloadSize);
+            }
```

Comment:
This should be done as a loop until end of payload rather than as nested method calls.

See https://github.com/reaktivity/nukleus-http.java/blob/develop/src/main/java/org/reaktivity/nukleus/http2/internal/stream/Http2ServerFactory.java#L2392